### PR TITLE
Remove promisify from event-emitter

### PR DESCRIPTION
**Pull request by qm3ster**
Simplifies the example

Transferred the promisify usage to `async` example instead

Also removed unused dependencies from `package.json`

### CODE REVIEW

File: event-emitter/lib/emitter.js [hunk 1]
```diff
@@ -30,38 +22,30 @@ class MyEventEmitter extends EventEmitter {
...
+        // Schedule the next iteration of the loop. This is performed with
+        // a `setImmediate` to yield to the event loop, to let JS code run
+        // and avoid a stack overflow.
+        setImmediate(loop);
+      });
     };
 
-    // Start the polling loop
-    loop();
+    // Start the polling loop on next iteration of the JS event loop to prevent zalgo.
```
Comment (kjvalencik) [inline event-emitter/lib/emitter.js]: ?

**Review by kjvalencik — COMMENTED** [review]
[deleted]

**Comment by qm3ster** [comment]
@kjvalencik I'm glad this will be fixed in core, this is an important usecase for me.
But I think we should provide the best possible solution in the meantime.

Do you think it would be harmful if I added the synchronous fetching of multiple events when available, as I described in the Slack reply thread?

File: event-emitter/lib/emitter.js (hunk 1 above)
Comment (amilajack) [inline event-emitter/lib/emitter.js:47]: Not everyone going through the examples will know what zalgo is. I think we should replace this with something more intuitive.

**Review by amilajack — CHANGES_REQUESTED** [review]
Some small changes but other than that, good stuff! Thanks for the contribution

File: event-emitter/lib/emitter.js (hunk 1 above)
Comment (qm3ster) [reply event-emitter/lib/emitter.js:47]: Agreed + I think this change actually isn't necessary the way the example is currently written.

**Review by amilajack — APPROVED** [review]

**Pull request merged**